pytest_mitmproxy.py

```diff
@@ -1 +1,76 @@
 # -*- coding: utf-8 -*-
+import re
+import os
+import shutil
+import pytest
+from contextlib import contextmanager
+from subprocess import Popen, CalledProcessError, TimeoutExpired
+from tempfile import TemporaryFile
+from urllib.parse import urlparse
+
+# clean the dumps each run
+if os.path.isdir("traffic_dumps"):
+    shutil.rmtree("traffic_dumps")
+
+
+@contextmanager
+def new_proxy(name, scope, flt=None):
+    ''' Start a new mitmdump process and
+    return the address for clients to connect
+
+    The captured content will be written to
+    ./<dump_dir>/<scope>/<name>
+    '''
+    dump_dir = "traffic_dumps"
+
+    os.makedirs("{}/{}".format(dump_dir, scope), mode=0o755, exist_ok=True)
+    dump_file = os.path.join(dump_dir, scope, name)
+
+    args = ["mitmdump", "--port=0", "--wfile={}".format(dump_file)]
+    args.append(flt) if flt else None
+
+    def read(fp):
+        fp.seek(0)
+        return fp.read().decode("utf-8")
+
+    with TemporaryFile() as stdout, TemporaryFile() as stderr:
+        ps = Popen(
```

Comment:
Actually yes; removed some other kwargs that were previously making it too long.